library/ssl_tls.c

```diff
@@ -297,6 +297,18 @@ int mbedtls_ssl_session_copy( mbedtls_ssl_session *dst,
     }
 #endif /* MBEDTLS_SSL_SESSION_TICKETS && MBEDTLS_SSL_CLI_C */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && \
+    defined(MBEDTLS_SSL_SESSION_TICKETS) && \
+    defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && \
+    defined(MBEDTLS_SSL_CLI_C)
+    if( src->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        dst->hostname = NULL;
```

Comment:
It might cause memory leak. and `session->hostname` has been inited as NULL.
```suggestion
```

Reply:
I don't think so.  Please see line 248.

Reply:
I prefer move it to mbedtls_ssl_session_free

Reply:
This has nothing to do with `mbedtls_ssl_session_free`. This is needed just because how the copy is done is this function. And it is out of the scope of this PR to change that.

Reply:
Wrong input. I have update it, it is due to shallow copy not session_free. So I prefer move it to line 245 , like `dst->ticket = NULL`

Reply:
Yes I agree better to move this as close as possible to the shallow copy (line 245). That will make things easier to understand.

Reply:
If we move the code to line 246, seems the code` "dst->ticket = NULL;"` can be removed.